src/nucleus/sanity.ml

```diff
@@ -77,6 +77,26 @@ let rec type_of_term_abstraction sgn = function
      let t_abstr = Abstract.abstraction Abstract.is_type a.atom_nonce t_abstr in
      Mk.abstract x t t_abstr
 
+let rec abstracted_judgement_with_boundary sgn abstr_jdg = 
+  match abstr_jdg with 
+    | NotAbstract (JudgementIsTerm jdg as jdg') -> 
+      Mk.not_abstract (jdg', BoundaryIsTerm (type_of_term sgn jdg))
+
+    | NotAbstract (JudgementIsType jdg as jdg')-> 
+      Mk.not_abstract (jdg', BoundaryIsType ())
+
+    | NotAbstract (JudgementEqType (EqType (asmp, t1, t2)) as jdg') -> 
+      Mk.not_abstract (jdg', BoundaryEqType (t1,t2))
+
+    | NotAbstract (JudgementEqTerm (EqTerm (asmp, e1, e2, t)) as jdg')-> 
+      Mk.not_abstract (jdg', BoundaryEqTerm (e1, e2, t)) 
```

Comment:
What is the invariant on `asmp` here? Are we sure we are allowed to form `BoundaryEqTerm` without attaching `asmp` to `e1`, `e2` or `t`?